File: corelibrary/src/main/java/pers/cxd/corelibrary/util/DataStructureUtil.java
package pers.cxd.corelibrary.util;

import java.util.Collection;
import java.util.Map;

/**
 * Miscellaneous {@code Collection} and {@code Map} and {@code Array} utility methods.
 *
 * @author pslilysm
 * @since 1.0.0
 */
public class DataStructureUtil {

    public static boolean isEmpty(Collection<?> collection) {
        return collection == null || collection.isEmpty();
    }

    public static boolean isEmpty(Map<?, ?> map) {
        return map == null || map.isEmpty();
    }

    public static boolean haveElement(Collection<?> collection) {
        return !isEmpty(collection);
    }

    public static boolean haveElement(Map<?, ?> map) {
        return map != null && !map.isEmpty();
    }

    public static boolean haveElement(boolean[] arr) {
        return arr != null && arr.length > 0;
    }

    public static boolean haveElement(byte[] arr) {
        return arr != null && arr.length > 0;
    }

    public static boolean haveElement(char[] arr) {
        return arr != null && arr.length > 0;
    }

    public static boolean haveElement(short[] arr) {
        return arr != null && arr.length > 0;
    }

    public static boolean haveElement(int[] arr) {
        return arr != null && arr.length > 0;
    }

    public static boolean haveElement(float[] arr) {
        return arr != null && arr.length > 0;
    }

    public static boolean haveElement(long[] arr) {
        return arr != null && arr.length > 0;
    }

    public static boolean haveElement(double[] arr) {
        return arr != null && arr.length > 0;
    }

    public static <T> boolean haveElement(T[] arr) {
        return arr != null && arr.length > 0;
    }

}

File: corelibrary/src/main/java/pers/cxd/corelibrary/util/DataStructureUtil.kt
package pers.cxd.corelibrary.util

/**
 * Miscellaneous `Collection` and `Map` and `Array` utility methods.
 *
 * @author pslilysm
 * @since 1.0.0
 */
object DataStructureUtil {
    @kotlin.jvm.JvmStatic
    fun isEmpty(collection: Collection<*>?): Boolean {
        return collection == null || collection.isEmpty()
    }

    @kotlin.jvm.JvmStatic
    fun isEmpty(map: Map<*, *>?): Boolean {
        return map == null || map.isEmpty()
    }
    @kotlin.jvm.JvmStatic
    fun haveElement(collection: Collection<*>?): Boolean {
        return !isEmpty(collection)
    }
    @kotlin.jvm.JvmStatic
    fun haveElement(map: Map<*, *>?): Boolean {
        return map != null && map.isNotEmpty()
    }
    @kotlin.jvm.JvmStatic
    fun haveElement(arr: BooleanArray?): Boolean {
        return arr != null && arr.isNotEmpty()
    }
    @kotlin.jvm.JvmStatic
    fun haveElement(arr: ByteArray?): Boolean {
        return arr != null && arr.isNotEmpty()
    }
    @kotlin.jvm.JvmStatic
    fun haveElement(arr: CharArray?): Boolean {
        return arr != null && arr.isNotEmpty()
    }
    @kotlin.jvm.JvmStatic
    fun haveElement(arr: ShortArray?): Boolean {
        return arr != null && arr.isNotEmpty()
    }
    @kotlin.jvm.JvmStatic
    fun haveElement(arr: IntArray?): Boolean {
        return arr != null && arr.isNotEmpty()
    }
    @kotlin.jvm.JvmStatic
    fun haveElement(arr: FloatArray?): Boolean {
        return arr != null && arr.isNotEmpty()
    }
    @kotlin.jvm.JvmStatic
    fun haveElement(arr: LongArray?): Boolean {
        return arr != null && arr.isNotEmpty()
    }
    @kotlin.jvm.JvmStatic
    fun haveElement(arr: DoubleArray?): Boolean {
        return arr != null && arr.isNotEmpty()
    }
    @kotlin.jvm.JvmStatic
    fun <T> haveElement(arr: Array<T>?): Boolean {
        return arr != null && arr.isNotEmpty()
    }
}